src/packages/views.py

```diff
@@ -91,37 +129,65 @@ def partner_new_order(request):
         destination_street_number = request.POST.get('destination_street_number')
         destination_address = get_address(destination_city, destination_area, destination_street, destination_street_number)
 
-        rate = float(request.POST.get('rate'))
+        rate = float(request.POST.get('rate').replace('₪', ''))
         phone_number = request.POST.get('phone_number')
+        full_name = request.POST.get('full_name')
 
-        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate, phone_number=phone_number)
+        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate,
+                               full_name=full_name, phone_number=phone_number)
 
     context = {
         'packages': order.related_packages(),
         'order': order,
         'partner': partner,
         'rates': partner.rates.split(','),
-        'cities': City.objects.all()
+        'cities': City.objects.all(),
+        'streets': Address.objects.values_list('street_name', flat=True),
+        'names': Package.objects.exclude(full_name='').values_list('full_name', flat=True)
     }
 
-    return render(request, 'packages/new_order.html', context)
+    return render(request, 'packages/order_edit.html', context)
 
 
-def get_address(city_name, area, street, street_number):
-    city = get_city(city_name, area)
-    address = Address.objects.filter(city=city, street_name=street, street_number=street_number)
+def package_edit_view(request, partner, order, package):
+    partner = Partner.objects.get(name=partner)
+    package = Package.objects.filter(id=package)
 
-    if not address:
-        return Address.objects.create(city=city, street_name=street, street_number=street_number)
-    return address.get()
+    if not is_member(request, partner.name) and not is_staff(request):
+        return render(request, 'errors/access_restricted.html', {})
 
+    context = {
+        'package': package.get(),
+        'order': order,
+        'partner': partner,
+        'rates': partner.rates.split(','),
+        'cities': City.objects.all(),
+        'streets': Address.objects.values_list('street_name', flat=True)
+    }
 
-def get_city(name, area):
-    city = City.objects.filter(name=name)
+    if request.POST:
+        origin_city = request.POST.get('origin_city')
+        origin_area = request.POST.get('origin_area')
+        origin_street = request.POST.get('origin_street')
+        origin_street_number = request.POST.get('origin_street_number')
+        origin_address = get_address(origin_city, origin_area, origin_street, origin_street_number)

```

Comment:
Here you might create an `Address`, but the `context['streets']` isn't updated.

Reply:
I missed that, stupid mistake (really should self review before opening this PRs...)